### PR TITLE
[TASK] Update to Psalm 5.6

### PR DESCRIPTION
**Pull request by oliverklee**

### CODE REVIEW

File: psalm.baseline.xml [hunk 1]
```diff
@@ -1,80 +1,20 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<files psalm-version="dev-master@">
+<files psalm-version="5.6.0@e784128902dfe01d489c4123d69918a9f3c1eac5">
```
Comment (JakeQZ) [inline psalm.baseline.xml:2]: We've seen this weird change before, and found it to be a non-issue, IIRC.

File: psalm.baseline.xml [hunk 2]
```diff
@@ -1,80 +1,20 @@
...
+            $cssRules['inlinable'],
+            /**
+             * @param array{selector: string, line: int} $first
+             * @param array{selector: string, line: int} $second
+             */
+            function (array $first, array $second): int {
+                return $this-&gt;sortBySelectorPrecedence($first, $second);
+            }
+        )</code>
+    </InvalidArgument>
```
Comment (JakeQZ) [inline psalm.baseline.xml:15]: We should investigate this as a separate issue.

File: psalm.xml [hunk 3]
```diff
@@ -2,6 +2,7 @@
 <psalm
   errorLevel="1"
   strictBinaryOperands="true"
+  findUnusedBaselineEntry="true"
```
Comment (JakeQZ) [inline psalm.xml:5]: Looks like a useful addition, though on just reading the [one-line description](https://psalm.dev/docs/running_psalm/configuration/#findunusedbaselineentry), I don't fully understand what it actually does.

**Review by JakeQZ — APPROVED** [review]
Interesting that a lot of the warnings have disappeared.  I guess the new version is smarter, and smart enough to determine that these are non-issues.

File: psalm.baseline.xml (hunk 2 above)
Comment (JakeQZ) [reply psalm.baseline.xml:15]: #1212

File: psalm.xml (hunk 3 above)
Comment (oliverklee) [reply psalm.xml:5]: This switch controls whether Psalm will complain about baseline entries listing a problem that does not exist (anymore).

**Comment by JakeQZ** [comment]
Oops; forgot to click squash and merge after approving.  Done now.

File: psalm.xml (hunk 3 above)
Comment (JakeQZ) [reply psalm.xml:5]: Ah, so some of the warnings that were removed in this PR should perhaps have been removed previously.

Comment (oliverklee) [reply psalm.xml:5]: Yes, possibly (though I tried to remember to update the baseline file when updating Psalm).

**Pull request merged**